packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -479,11 +479,19 @@ export interface SdkQueryParameter extends SdkModelPropertyTypeBase {
   collectionFormat?: CollectionFormat;
   serializedName: string;
   correspondingMethodParams: SdkModelPropertyType[];
+  explode?: boolean;
 }
 
 export interface SdkPathParameter extends SdkModelPropertyTypeBase {
   kind: "path";
+  /**
+   * @deprecated This property is deprecated. Use `allowReserved` instead.
+   * @param serviceParam
+   */
   urlEncode: boolean;
+  explode?: boolean;
+  style?: "simple" | "label" | "matrix" | "fragment" | "path";
+  allowReserved?: boolean;
```

Comment:
Is `allowReserved` only for path parameter? Can query parameter define this? I did not find it is only applied to path parameter in [RFC](https://datatracker.ietf.org/doc/html/rfc6570#page-22) and [issue](https://github.com/microsoft/typespec/issues/3736).
<img width="568" alt="image" src="https://github.com/user-attachments/assets/1e65c229-a0c1-4268-bd99-eba23ced649f">

Reply:
i infer this conclusion from typespec's design. @timotheeguerin could you confirm?

Reply:
Yeah query params cannot but I think after some investigation is is kinda pointless as to skip encoding for query params

Reply:
@timotheeguerin, in the RFC there are examples that `allowReserved` can be applied to query parameters. May I know why query params cannot `allowReserved` and it is pointless? 
https://datatracker.ietf.org/doc/html/rfc6570#page-22
<img width="560" alt="image" src="https://github.com/user-attachments/assets/e4b39b7b-7d5b-4ea4-9679-cd60e9868576">

/cc @srnagar

Reply:
@timotheeguerin we have query with `"x-ms-skip-url-encoding": true` in [swagger](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/web/resource-manager/Microsoft.Web/stable/2020-06-01/Recommendations.json#L476-L481). I guess for service team side this will be helpful in paging filter in which customers would input the already encoded string.